tests/test_disable_tf_exact_match_detection.py

```diff
@@ -0,0 +1,279 @@
+
+    df_predict = linker.predict()
+
+    sql = f"""
+    select * from {df_predict.physical_name}
+    where unique_id_l = 835
+    and unique_id_r = 836
+    """
+    res = linker.query_sql(sql).to_dict(orient="records")[0]
+
+    # Exact match, normal tf adjustement, Kirk
+    assert res["bf_surname"] == pytest.approx(8.0)
+    # Overall BF should be m/u = 0.8/0.2 = 4
+    assert res["bf_tf_adj_surname"] * res["bf_surname"] == pytest.approx(4.0)
+
+    # Levenshtein match, normal tf adustments
+    sql = f"""
+    select * from {df_predict.physical_name}
+    where unique_id_l = 147
+    and unique_id_r = 975
+    """
+    res = linker.query_sql(sql).to_dict(orient="records")[0]
+    # Levenshtein match, normal tf adustments, Taylor
+    # Splink makes the tf adjustment based on on the exact match level
+    # Lev match level has bf of 0.9/0.3
+    assert res["bf_surname"] == pytest.approx(3.0)
+    # Overall BR should be based base of 3.0
+    # TF adjustmeent is difference between:
+    #    u of exact match = 0.1
+    #    u of Taylor = 0.4
+    assert res["bf_surname"] * res["bf_tf_adj_surname"] == pytest.approx(
+        3.0 * (0.1 / 0.4)
+    )
+
+    settings_disabled = get_settings(disable_tf_exact_match_detection=True)
+    linker = Linker(df, settings_disabled, DuckDBAPI())
+
+    tf_lookup = [
+        {"surname": "Taylor", "tf_surname": 0.4},
+        {"surname": "Kirk", "tf_surname": 0.2},
+    ]
+    linker.register_term_frequency_lookup(tf_lookup, "surname")
+
+    df_predict = linker.predict()
+
+    sql = f"""
+    select * from {df_predict.physical_name}
+    where unique_id_l = 835
+    and unique_id_r = 836
+    """
+    res = linker.query_sql(sql).to_dict(orient="records")[0]
+    # Exact match, normal tf adjustement, Kirk
+    assert res["bf_surname"] == pytest.approx(8.0)
+    # Overall BF should be m/u = 0.8/0.2 = 4
+    assert res["bf_tf_adj_surname"] * res["bf_surname"] == pytest.approx(4.0)
+
+    sql = f"""
+    select * from {df_predict.physical_name}
+    where unique_id_l = 147
+    and unique_id_r = 975
+    """
+    res = linker.query_sql(sql).to_dict(orient="records")[0]
+    # Levenshtein match, tf exact match detection disabled, Taylor
+    # Splink makes the tf adjustment based on on the exact match level
+    # Lev match level has bf of 0.9/0.3
+    assert res["bf_surname"] == pytest.approx(3.0)
+
+    # Overall BR should be based base of 3.0
+    # TF adjustmeent is difference between:
+    #    u of this level = 0.3
+    #    u of Taylor = 0.4
+    assert res["bf_surname"] * res["bf_tf_adj_surname"] == pytest.approx(
+        3.0 * (0.3 / 0.4)
+    )
+
+    settings_disabled_with_min_tf = get_settings(
+        disable_tf_exact_match_detection=True, tf_minimum_u_value=0.1
+    )
+
+    linker_base = Linker(df, settings_disabled_with_min_tf, DuckDBAPI())
+    linkers = [linker_base, Linker(df, linker_base.save_model_to_json(), DuckDBAPI())]
```

Comment:
This also tests that [this](https://github.com/moj-analytical-services/splink/pull/2122) PR has been correctly incorporated